cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -107,30 +132,80 @@ function mapDispatchToProps(dispatch: any): DispatchToProps {
 
 type Props = StateToProps & DispatchToProps;
 class AnnotationTopBarContainer extends React.PureComponent<Props> {
+    private autoSaveInterval: number | undefined;
+
     public componentDidUpdate(): void {
         const {
             jobInstance,
             frameNumber,
+            frameChangeTime,
+            frameSpeed,
             playing,
             canvasIsReady,
-            onChangeFrame,
             onSwitchPlay,
         } = this.props;
 
+        
         if (playing && canvasIsReady) {
             if (frameNumber < jobInstance.stopFrame) {
+                const delay: number = frameChangeTime ? Math.max(0,~~(1000/frameSpeed) -  new Date().getTime() + frameChangeTime) : 0;
                 setTimeout(() => {
                     const { playing: stillPlaying } = this.props;
                     if (stillPlaying) {
-                        onChangeFrame(frameNumber + 1);
+                        this.onChangeFrame(frameNumber + 1, new Date().getTime());
```

Comment:
If we trying to calculate delay depends on updating time I would suggest simpler method:

``changeFrameAsync`` gets time at the beginning and at the end of the function. It computes difference. 
Finally it pushes difference to store via the CHANGE_FRAME_SUCCESS action.
The store uses this value to update ``delay`` value (using ``frameSpeed`` as a base value).
The top bar container gets ``delay`` value from the state and just uses it here. 

It would allow us remove extra argument from ``onChangeFrame`` and incapsulate all logic related with computing time in the one place.

Reply:
> `changeFrameAsync` gets time at the beginning and at the end of the function. It computes difference.

Such approach can't satisfy applicable timing as i can imagine, because there are some computation out of the `changeFrameAsync` and delay can't be correctly computed. But we can still get time one time somewhere in `changeFrameAsync` and put in a state and compute delay using previous one.